### ISSUE TITLE
Fix false positives shouldForwardProp in property-no-unknown

### ISSUE THREAD

**Issue by alexandrzavalii** [stylelint/stylelint#4639]
Stylint throws an error when using shouldForwardProp from styled-system

> What code is needed to reproduce the bug?

```
import styled from '@emotion/styled'
import shouldForwardProp from '@styled-system/should-forward-prop'

export const Box = styled('div', { shouldForwardProp })({
    boxSizing: 'border-box',
  },
  props => props.css,
)
```

> What stylelint configuration is needed to reproduce the bug?

```
    "@emotion/core": "10.0.27",
    "@emotion/styled": "10.0.27",
    "@styled-system/should-forward-prop": "5.1.5",
    "styled-system": "5.1.5",
    "stylelint": "13.2.0",
```

```
module.exports = {
  extends: ['stylelint-config-recommended', 'stylelint-config-rational-order'],
  ignoreFiles: ['**/storybook-static/**'],
  rules: {
    'at-rule-no-unknown': null,
    'plugin/rational-order': [
      true,
      {
        'border-in-box-model': true,
        'empty-line-between-groups': false,
      },
    ],
  },
};

```

**Comment by jeddy3** [comment]
@alexandrzavalii Thanks for the report.

I can [reproduce this with the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICWBbADhATgFwAQGdsBPAGxilwDNMIVcByAARhQmyQgDsB6Q08+gB1OqDDgIALCAFcSUAGJYA7gENMUAAo00VGnSZ8yUALT4ihFrymyTlZWpNptQzsJgAPMXkidCuAEIQ7rgAvATERgAU9FBIAG70ADS4wJIycoqYqupaEDoAvgCUkcDCuOW4AEZBAMpIAF5InADm8AzV6jCYxtXuSWW4+YkDTnn4oQB8uKNo+AB0YPj4w5yFIIkgZpzYKu4I4EvGTcYAVvjr4FyUSM37pZzlgiCYsjD4T233FbhPKtjGLzIxk4EGM0k4AGsQUpXAhcNgXjAVt8npUSBAwBDgaCWGhiB94YjkRUnpB0d0QUdOHEVCQkCYJB4CQjpEiBo9LigUDBttjjLj8XCWWyHiSQLAwCQ1H8OJweujMXyoNI0HSwH8YMYZl12G8CQBtdkVYXE75fb5im4gzAwA1Gi0-S6+GBgaTsOKa5WqpDq7BvYxKJDYCTGWKUShdHn-GkkVnvED2ioAXUT+SNSdNjolUswMq48oxWMp+CkOAkKk4jm0OqIxggHswmHpeqFRKNTzsvMoKhQSBItcpXrVGuBPZbbRN7ZAnf+3d7-b5vaWTWaxmaPK6PuMc77tYhMCISiwUGZbdFHMo4LA7Hz6pIYD5TRj9NPrMzHavN7ld4flPB+DQFQwHIOs0C6P4sFfEUUWnT9ZWMOlOBgNQ11zWIoz5EFfB2SsHFDJAbWvWUoPfEB9yIagx1DF0czzOVKVELAcOwEip25WIVG3ZDsGkG1R25PlwShCAYVY89HUHN5gMrFdTDAsAkGuBTBQnM8YMHFVhz9Yw-iOdAmIBV541Ut8pw071fU1fAYDIa8sGMwlTPEp5KQFWt8BkTBgLE9ScXcBFOOs3syS4BzJ2ckAGOpWl6VDGQ0SsqUS2MSAuSjMK1LFbUcAHUEhOhWETOgsVrNs7AsC1azpCgUFJRUQ4-0hAqfJKmyXXK7pZhgaqcTIbleUa4TRNbJyYNK9qKuIMDBKakTCsc4qOUIJsWkwmAlEQ20RsWx1wSDGahvm8KYN0wFNTchCmk1SoYDsG07XEjlaVUcwnkzcpzQdJ4PGAvEHodGC0ULMhDhUSg-W6fAqM4KigcxEH43ei0OwIwhgTeP0TwTR7kyRjkrSwLa2kNHGTvBroUtofqWOxgHcBTHG03EjMp1O15+M1dVrIJJ50SULo3tZ-4zo5uSgM1MGIZ5kBnpUcxTBXIFNsFiK2aBIKfQgdF6PWzadPJzBpdl171inOGsUlCB8Fkypc2A-l0GIS6kJ6W7CelpCGxVwGFQt9FrdW23xfRjarv1qW4V+EgXsRs3fZS-2bbtzUkNDl2bruonHWN-BjBQWR2GdrbSPNhOraT4OAODjP3cjmXo7l3OA+aJWru9sVS7yHkK-t1O9clgW65zvOC6QIv245TuwJEQPk7F+2B8NoeG-l5vW6QifHSn7vZ8rwD7Zr+7l5jzfSS1irGXcFL6qz3mRMH0iQu6S+EJ5Zpg2lksmNPzlqYdvFax60PlnEmX164n1NjjT6yMQA-RgH9OEoC6aOmuJgNGSELBY0TOUFmpNHQEyPsTbBJUIg2SaP8VKKAKxQHjMQhmDombfFwTBShGElASCDG8femomjW1gEbFescIqukILQLU1Ycr-ydkAt2hDcBIJOoIzeH1iHfXcL9GmRC8FPQNilaQoiUDiK7jlZRMDUHoIxuQJ4dC8b4OaNaEBxCdEQ0pmlbYpiYK8ObArFo69rq+2sXg+hFpGG4ynNmaUX4egVlXFXBeBsPYwC9pAmCETcxRMqDE+e11ZG33AY3H+aS6IFkVBrEKOs04SwScfRuI8SCF2ViksURSMnxzKefOUcSqkRzaFHGOPiW6akaaRFp8FS7tO1tk12mdEnJJGTRSJYy2mKyGWHKhV9RmhQJAARnmXVdJSzCzGACn2WSEyuACIgXs2iUTykh37tU3p+T5b53qWPYZ4SFkHNvB0qZi9Lm1LXqsjeTSOSbO-L8rp0za5PM9g-T5+zikXRkTMxBiY+kFNBRUaBME4EIK0cg7OujwUeLFOY-4GDMaBLpswgGTwCGOO0USlxrD3FYrAV42AAy-ElIhNSgGwTvihJwaRS8nAiI-K5JxPuYd-k1JeaPceWKPzituVTQK3Dw6DyecPIFSrRVwUlVQqZwDZnwoimKiVcp1lSMAVdByAAGA1qr4IwwElzPJfNzUwUtVEwCNptiMmsrnGVLtOV5OHq8hpbdlWwRdfmf1UYg3+iheGs1kMVn6qnL6+C7DOFQrlTqpRsamiwG2HRAkAAmUiZAWjBmMPULooJGpBhahyG1yL7U7NIuxJAnEIx-F4pzSFmrC3Z2LT28gfauKDr4ncqFpq65wqXpOjiM6eJ8TdZzG+0svUrrYlO-t3Eh1ajUEmt4KbNVpqXUk71Ype1HtnZqXMLRNTGIgpDUdjzx1XIPWugdG7n0xLfWBdJFUF25IBSbVd06ACOrJMB2rRpQ6VutZXfoxQqt5WaIoPuMPBroSGKHqr+Rh55TdM0fNw4e-DCGiOuONRB1FsLb37oiq5LWAYOF+i6W2iSoIlzN35JWOslBTCeV7mhkFRVSKcGkCgG63QyAqFiKtRtNAoNCPUvJxTfITmIVXOpq2fGnjZSdh63d982MwRFp29OkG0U40w7nKN7yY1IxxWKPFmj5FOJQajCllisFBNsfS+xMLfNMt+Lo1lNM6GpnTKRcadluh-AREgSobprp233NgJuV7fDNnTT-ZLHUdLYHS5l7S76ysFrI8ukrbUUvlcq1l0C4FauasXSxuZU5SsVVSpUJoH7SM9J-ZipLTWyuDeG516uDmi2-oiv17osB8DSSgBWYjCnZsVUpFhbJJnNhTYqnSZDJHQ3dO1eN6DfWTtKaQOdqVJqFsSVY41sqFUO2OztRggkTq7ufc6lVGqCd6q5ws3XPdH2JrA+6qDuqhxE2Bovfl4O16et3qWvdyq8PeosAwpDp50PY0rdxz1fkfVCe-JBDsL80sarSASjD5rU1t3cyh1Zn+e0KE7s5-zazYoYysmMORI86hr4c+J1z2NwvVlPeNZdrVS9Fu1NczhmCcvLoK41cHMdznuXAuLlOLXZ3tuMa669ly73Ze0hF2bvOuxbVF0dQMJm+QQD5CAA).

I believe it's an issue with the syntax that [stylelint uses to support CSS-in-JS](https://stylelint.io/about/syntaxes). Please consider [contributing to the syntax](https://stylelint.io/developer-guide/syntaxes#working-on-syntaxes) if you have time.

I've added this issue to the [CSS-in-JS umbrella issue](https://github.com/stylelint/stylelint/issues/4574).

**Comment by AhmedBHameed** [comment]
I confirm the same issue here
![issue](https://user-images.githubusercontent.com/25675361/113436961-07f21980-93e6-11eb-8f1d-1d672152d661.png)

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

We'll also [be deprecating](https://github.com/stylelint/postcss-css-in-js/issues/225) our [forked CSS-in-JS package](https://github.com/stylelint/postcss-css-in-js), in favour of leaner custom syntaxes.

Please consider [writing a custom syntax](https://github.com/stylelint/stylelint/blob/main/docs/developer-guide/syntaxes.md) specific to the CSS-in-JS library in this issue and fixing the bug there.
